spec/index.html

```diff
@@ -2686,8 +2693,246 @@ <h3>Selective Disclosure Schemes</h3>
+        `_:c14n1 &lt;http://schema.org/[f-g]`, which helps us guess that the hidden predicate is
+        `&lt;http://schema.org/gender&gt;` with high probability. Alternatively, we can assume that the guesser already has
+        such knowledge via the public credential schema.</p>
+      
+      <p>Then, if the canonical labeling produces different results depending on the gender value, we can use it to deduce the
+        gender value. In fact, this example produces different results depending on whether the gender is `Female` or `Male`.
+        (Note: ignored the other types of gender just for brevity)</p>
+      
+      <p>The following example shows that `gender` = `Male` yields different canonical labeling.</p>
+
+      <pre id="ex-pc-leakage-labeling-hypothetical-normalized-dataset" class="example" data-transform="updateExample" title="Hypothetical Normalized Dataset">
+        <!--
+          _:c14n0 <http://schema.org/address> _:c14n1 .
+          _:c14n0 <http://schema.org/familyName> "Jarrett" .
+          _:c14n0 <http://schema.org/gender> "Male" .  # gender === Male
+          _:c14n0 <http://schema.org/givenName> "Ali" .
+          _:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person> .
+          _:c14n1 <http://schema.org/addressCountry> "United States" .
+          _:c14n1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/PostalAddress> .
+        -->
+      </pre>
+
+      <p>So the verifier should have obtained the following dataset if `gender` had the value `Male`, which differs from the
+        revealed dataset. Therefore, the verifier can conclude that the `gender` is `Female`.</p>
+
+      <pre id="ex-pc-leakage-labeling-hypothetical-disclosed-dataset" class="example" data-transform="updateExample" title="Hypothetical Disclosed Dataset">
+        <!--
+          _:c14n0 <http://schema.org/address> _:c14n1 .
+          _:c14n0 <http://schema.org/familyName> "Jarrett" .
+          ########### 3rd statement is unrevealed ##########
+          _:c14n0 <http://schema.org/givenName> "Ali" .
+          _:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person> .
+          _:c14n1 <http://schema.org/addressCountry> "United States" .
+          _:c14n1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/PostalAddress> .
+        -->
+      </pre>
+
+      <p>Note that we can use the same approach to guess non-boolean values if the range of possible values is still a
+        reasonable (small) size for a guesser to try all the possibilities.</p>
+
+      <p>By making the canonicalization process private, we can prevent a brute-forcing attacker from trying to see the
+        labeling change by trying multiple possible attribute values.
+        For example, we can use a HMAC instead of a hash function in the canonicalization algorithm. Alternatively, we can add
+        a secret random nonce (always undisclosed) into the dataset.
```

Comment:
This is a similar remark, with the difference that we have not, in my recollection, discussed to add a nonce to the algorithm (I presume to be used with the hash function). I think it would be possible and probably worthwhile to allow that, but that would require a separate WG resolution (and further test cases, b.t.w.)

Reply:
It may be better to just make it clear that other specs may parameterize the hash function using whatever they like -- noting that it will not longer be URDNA2015 (or whatever we end up calling it), but instead some variant.

Reply:
Naming aside, I would expect all parameters would have a default, probably using SHA256 for hash, no nonce, and sorting just the good old unicode sorting as now. Ie, if we do these additions with defaults we would not harm deployed code (I presume that is your worry, @dlongley). Defining another spec that does exactly the same but parametrizing the hash function sounds a bit of an overkill to me...